src/glod/in_out/mail_merge/chy3_merge.py

```diff
@@ -13,82 +13,120 @@
 
 LOG = logging.getLogger(__name__)
 
-DONOR = "Name of donor"
-PPS = "PPS Number"
-POSTAL_ADDRESS = "undefined"
-PHONE = "Phone no"
+DONOR = "Name of Donor"
+PPS = "PPSN"
+POSTAL_ADDRESS = "Address incl. Eircode"
+PHONE = "Phone No"
 EMAIL = "Email Address"
-CHARITY_NAME = "Name of eligible charity or other approved body hereinafter referred to as an approved body"
-VALID_FROM = "Text20"
-
-
-def _merge_letters(gdrive, gdocs, temp_dir, template_file_id, template_filename, valid_from_tax_year, targets):
-    for selected_member, person, household_org, address in targets:
-        LOG.info(f"Merging letter for {person.name}, household id: {household_org.id}")
-        cover_letter_filename = f"cover.{household_org.id}.pdf"
+CHARITY_NAME = "Name of eligible charity or other approved body"
+VALID_FROM = "0"
+
+
+def _merge_letters(
+    gdrive,
+    gdocs,
+    temp_dir,
+    template_file_id,
+    template_filename,
+    valid_from_tax_year,
+    targets,
+):
+    for given_name, donor, pps, postal_address, phone, email, household_id in targets:
+        LOG.info(f"Merging letter for {given_name}, household id: {household_id}")
+        cover_letter_filename = f"cover.{household_id}.pdf"
         cover_letter_path = os.path.join(temp_dir, cover_letter_filename)
-        replacements = dict(given_name=person.given_name)
-        with merge_letter(gdrive, gdocs, template_file_id, replacements) as merged_file_id:
+        replacements = dict(given_name=given_name)
+        with merge_letter(
+            gdrive, gdocs, template_file_id, replacements
+        ) as merged_file_id:
             download_as_pdf(gdrive, merged_file_id, cover_letter_path)
 
-        chy3_filename = f"chy3.{household_org.id}.pdf"
+        chy3_filename = f"chy3.{household_id}.pdf"
         chy3_file_path = os.path.join(temp_dir, chy3_filename)
         merge_data = {
-            DONOR: selected_member.name_override if selected_member.name_override else person.name_without_title,
-            PPS: selected_member.pps if selected_member.pps else '',
-            POSTAL_ADDRESS: address.post_label(),
-            PHONE: person.mobile if person.mobile else address.telephone,
-            EMAIL: person.email,
+            DONOR: donor,
+            PPS: pps,
+            POSTAL_ADDRESS: postal_address,
+            PHONE: phone,
+            EMAIL: email,
             CHARITY_NAME: configuration.charity.name,
             VALID_FROM: str(valid_from_tax_year),
         }
         fill_form(template_filename, chy3_file_path, merge_data)
 
-        output_file = os.path.join(temp_dir, f"combined.{household_org.id}.pdf")
+        output_file = os.path.join(temp_dir, f"combined.{household_id}.pdf")
         concatenate([cover_letter_path, chy3_file_path], output_file)
 
         yield output_file
 
 
-def merge_chy3_letters(template_filename):
+def merge_chy3_letters(
+    template_filename, input_workbook_file_id, sheet_name, template_file_id
```

Comment:
empty_certificate_form, input_workbook_file_id, sheet_name, template_letter_file_id

and rename calling code